docs/DeveloperGuide.md

```diff
@@ -1,34 +1,182 @@
-# Developer Guide
+# Control Your Crowd - Developer Guide
 
-## Design & implementation
+* [Setting up](DeveloperGuide.md#setting-up)
+* [Design](DeveloperGuide.md#design)
+  * [Architecture](DeveloperGuide.md#architecture)
+  * [UI component](DeveloperGuide.md#ui-component)
+  * [Logic component](DeveloperGuide.md#logic-component)
+  * [Model component](DeveloperGuide.md#model-component)
+  * [Storage component](DeveloperGuide.md#storage-component)
+  * [Common classes](DeveloperGuide.md#common-classes)
+* [Implementation](DeveloperGuide.md#implementation)
+* [Documentation](DeveloperGuide.md#documentation)
+* [Testing](DeveloperGuide.md#testing)
+* [Dev Ops](DeveloperGuide.md#dev-ops)
+* [Appendix](DeveloperGuide.md#appendix)
+  * [Appendix A: Product Scope](DeveloperGuide.md#appendix-a-product-scope)
+    * [Target user profile](DeveloperGuide.md#target-user-profile)
+    * [Value proposition](DeveloperGuide.md#value-proposition)
+  * [Appendix B: User Stories](DeveloperGuide.md#appendix-b-user-stories)
+  * [Appendix C: Use Cases](DeveloperGuide.md#appendix-c-use-cases)
+  * [Appendix D: Non Functional Requirements](DeveloperGuide.md#appendix-d-non-functional-requirements)
+  * [Appendix E: Glossary](DeveloperGuide.md#appendix-e-glossary)
+  * [Appendix F: Instructions for Manual Testing](DeveloperGuide.md#appendix-f-instructions-for-manual-testing)
+    * [Launch and Shutdown](DeveloperGuide.md#launch-and-shutdown)
+    * [Clearing the list](DeveloperGuide.md#clearing-the-list)
+    * [Saving data](DeveloperGuide.md#saving-data)
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+## Setting up
 
+## Design
 
-## Product scope
-### Target user profile
+### Architecture
```

Comment:
It's good to contain the architecture section. Maybe it can be better to provide a diagram to show the overall architecture. ( Refer to the AB3 DG example)